src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,105 @@
+
+		private IServiceFabricClient? m_client;
+
+		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
+
+		private string? m_nodeName;
+
+		internal static new string? HealthReportSourceId => nameof(RestHealthCheckPublisher);
+
+		internal const string NodeNameVariableName = "Fabric_NodeName";
+
+		public RestHealthCheckPublisher(ILogger<ServiceFabricHealthCheckPublisher> logger,
+										IOptions<RestHealthCheckPublisherOptions> options,
+										ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
+		{
+			m_nodeName = FindNodeName();
+			m_clientWrapper = new(new Uri(options.Value.RestHealthPublisherClusterEndpoint));
+			m_logger = logger;
+		}
+
+		public override async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+		{
+			if (m_nodeName == null)
+			{
+				m_logger.LogError(Tag.Create(), "Can't find node name.");
+				return;
+			}
+
+			m_client = await m_clientWrapper.GetAsync();
+
+			Action<ServiceFabricHealth.HealthInformation> reportHealth =
+				async (sfHealthInfo) => await PublishHealthInfoAsync(sfHealthInfo);
+
+			PublishAllEntries(report, reportHealth, cancellationToken);
+		}
+
+		internal static ServiceFabricCommon.HealthInformation FromSfHealthInformation(ServiceFabricHealth.HealthInformation healthInfo)
+		{
+			ServiceFabricCommon.HealthState healthState = healthInfo.HealthState switch
+			{
+				ServiceFabricHealth.HealthState.Invalid => ServiceFabricCommon.HealthState.Invalid,
+				ServiceFabricHealth.HealthState.Ok => ServiceFabricCommon.HealthState.Ok,
+				ServiceFabricHealth.HealthState.Warning => ServiceFabricCommon.HealthState.Warning,
+				ServiceFabricHealth.HealthState.Error => ServiceFabricCommon.HealthState.Error,
+				ServiceFabricHealth.HealthState.Unknown => ServiceFabricCommon.HealthState.Unknown,
+				_ => throw new ArgumentException($"'{healthInfo.HealthState}' is not a valid health status."),
+			};
+
+			ServiceFabricCommon.HealthInformation sfcHealthInfo = new(
+				   sourceId: healthInfo.SourceId,
+				   property: healthInfo.Property,
+				   healthState: healthState,
+				   timeToLiveInMilliSeconds: healthInfo.TimeToLive,
+				   description: healthInfo.Description,
+				   sequenceNumber: healthInfo.SequenceNumber.ToString(),
+				   removeWhenExpired: healthInfo.RemoveWhenExpired,
+				   healthReportId: healthInfo.HealthReportId
+		   );
+
+			return sfcHealthInfo;
+		}
+
+		internal async Task PublishHealthInfoAsync(ServiceFabricHealth.HealthInformation healthInfo)
+		{
+			ServiceFabricCommon.HealthInformation sfcHealthInfo = FromSfHealthInformation(healthInfo);
+
+			if (m_client == null)
+			{
+				m_logger.LogInformation(Tag.Create(), "HTTP Client isn't ready yet.");
+				return;
+			}
+
+			await m_client.Nodes.ReportNodeHealthAsync(
+				nodeName: m_nodeName,
+				healthInformation: sfcHealthInfo);
+		}
+
+		internal string? FindNodeName() => Environment.GetEnvironmentVariable(NodeNameVariableName);
```

Comment:
How will this variable be set?

Reply:
I'm not sure, what do you mean?

Reply:
This variable provided by SF, in the future it would be moved to IExecutionContext https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-environment-variables-reference#:~:text=Service%20Fabric%20environment%20variables%20%20%20%20Environment,%20%208234%20%2011%20more%20rows%20

Reply:
Might be worth adding a code comment with that info